src/vivarium_census_prl_synth_pop/components/immigration.py

```diff
@@ -13,21 +14,22 @@
 )
 
 
-class Immigration:
+class Immigration(Component):
     """
     Handles migration of individuals *into* the US.
     """
 
-    def __repr__(self) -> str:
-        return "Immigration()"
-
     ##############
     # Properties #
     ##############
 
     @property
-    def name(self):
-        return "immigration"
+    def columns_required(self) -> List[str]:
+        return []
```

Comment:
Does this component needs all columns in its population view? I don't see a population view at all in the old code.

Reply:
I was unsure of this. There is no population view but doesn't the component still need to use columns?